src/ORM/FieldType/DBLocale.php

```diff
@@ -2,13 +2,18 @@
 
 namespace SilverStripe\ORM\FieldType;
 
+use SilverStripe\Core\Validation\FieldValidation\LocaleFieldValidator;
 use SilverStripe\i18n\i18n;
 
 /**
  * Locale database field
  */
 class DBLocale extends DBVarchar
 {
+    private static array $field_validators = [
+        LocaleFieldValidator::class,
+    ];
+
     public function __construct(?string $name = null, int $size = 16)
```

Comment:
I don't know? Can you point to what that might be?

Reply:
> Can you point to what that might be?

To what the reason this is a size 16 is, or to what the mismatch is?
I don't know why this is size 16, but the mismatch is that all of the valid locales are only up to 5 characters long. So we're saying "it can be 16 characters long but only if it's one of these 2 to 5 character strings". Which indicates that either the validator isn't allowing all of the intended values, or the size is wrong.

Reply:
:shrug: We're using a standardised symfony locale validator to validate a locale.  Seems like that's the best way of determining what a valid locale is?

There's little in the way of Silverstripe specific locale docs, SilverStripe\i18n\Data\Locales has some doc blocks with 5 character locale examples. https://docs.silverstripe.org/en/5/developer_guides/i18n/ has 5 character locale examples, so does https://docs.silverstripe.org/en/5/optional_features/elemental/advanced_setup/

I think we should just leave the default 16 size as it is so we don't accidentally truncate someones existing data on migration on the off chance there is something we haven't considered, and loosen the validator later on if there is community feedback

Reply:
That seems like it's really misleading at best and like we're ignoring a potential regression at worst but sure.